src/wovnio/wovnphp/Url.php

```diff
@@ -150,21 +149,64 @@ private static function addQueryLangCode($uri, $lang_code, $lang_param_name)
     }
 
     /**
-     * Public function removing the lang of the url
+     * Adds a lang code to a URL with no lang code, using path pattern.
+     * Supports both absolute and relative paths.
+     * @param $no_lang_url
+     * @param $lang
+     * @param string $site_prefix_path
+     * @return string|string[]|null
+     */
+    private static function addPathLangCode($no_lang_url, $lang, $site_prefix_path = '')
+    {
+        if (empty($lang)) {
+            return $no_lang_url;
+        }
+        return preg_replace(
+            self::generateUrlRegex($site_prefix_path),
+            "$1$2$3/$lang$4",
+            $no_lang_url
+        );
+    }
+
+    /**
+     * Adds lang code to a URI following the subdomain pattern.
+     * No custom lang alias conversion happens here.
      *
-     * @param String $uri The url with the lang
-     * @param String $pattern
-     * @param String $lang_code The lang to remove
-     * @return array The url without the lang
+     * @param $parsed_url array parse_url output array
+     * @param $lang_code String language code
+     * @param $no_lang_uri String URI without language code
+     * @return string|string[]|null
      */
-    public static function removeLangCode($uri, $lang_code, $settings)
+    private static function addSubdomainLangCode($parsed_url, $lang_code, $no_lang_uri)
     {
-        if (!$lang_code || strlen($lang_code) == 0) {
-            return $uri;
+        // check if subdomain already exists
+        if (preg_match('/.*\..*\..*/', $parsed_url['host'])) {
```

Comment:
Not too sure about this one. Is this making assumptions about how many subdomains there are? `a.b.c.d.com` is valid, for example. Maybe look at how we do it in html-swapper and other libraries.

Reply:
That's a good point, totally also didn't consider the case where people having `company.co.uk` kind of domains. Will fix.